lib/src/layer/tile_layer/tile_image_view.dart

```diff
@@ -9,11 +9,11 @@ class TileImageView {
   final DiscreteTileRange _visibleRange;
   final DiscreteTileRange _keepRange;
 
-  TileImageView({
+  const TileImageView({
```

Comment:
These java annotations have no meaning other than expressing  constraints on sub-types. Please do not inherit from TileImageView.

I could mark it @final but if sub-typing is a concern I'd rather kill the file and make it an internal class to the tile manager.

Reply:
While it doesn't give any performance improvements, I won't hurt performance too. 
Immutable annotations helps to mark model classes and enforce their intended use in the future through lint warnings.
In my opinion they are nice to have, especially with lint checking in the CI.

Reply:
I'm all for explicit intent that's why I'm saying @immutable is a poor choice.

The fact that you're arguing that there can be linter warnings when someone inherits from TileImageView makes me wonder if you think folks should inherit from TileImageView? If I saw an @immutable annotation, I certainly would wonder if that's what the authors wanted me to do.

I've offered you more explicit alternatives, i.e. squash the file and move TileImageView as _TileImageView into the manager's file.

Reply:
I wouldn't particularly mind either way (I don't do it for my own projects), but for consistency it's probably worthwhile.

I think what @josxha's trying to say is that if you add a non-final attribute to an \@immutable marked class (for example in another PR), then the linter will immediately get you to reconsider. Less so for others inheriting in their own code. But I also don't know whether \@final does this as well.

Reply:
Sorry if I sound reluctant. I wanna be a good sport. I just don't believe in following the process for the sake of process. If anyone has an argument on why the immutable annotation should be there, I'll add it in a heartbeat, no questions asked.

> I think what @josxha's trying to say is that if you add a non-final attribute to an https://github.com/immutable marked class (for example in another PR), then the linter will immediately get you to reconsider. Less so for others inheriting in their own code.

The linter will do that for you for any class with a const constructor. The immutable annotation is specifically to enforce constraints on subtypes: https://api.flutter.dev/flutter/meta/immutable-constant.html.

I'm all for consistency, if this was a class that should be sub-typed and other classes/interfaces that should be sub-typed were annotated as well, I would annotate this class. Unfortunately, neither holds here and especially nobody should feel encouraged to subtype this class.

It's not even clear to me, what a good use-case for @Immutable would look like. Let's say, we defined a base-class for folks to extend. Why would we disallow our users, the implementers, to have a constructor or have mutable members for caching? Maybe I could imagine a case, where we'd manage a complex lifecycle of the object and we wouldn't want implementation to implicitly depend on it :shrug: 

I added a final class modifier now, which will at least disallow sub-typing from outside this package. I still think the most consistent would be to make it a private class, since users should neither sub-type it nor look at it.

(and completely tangentially, I believe that as we push performance further and further, for any GC optimizatoins constness/immutably is here to bite us. Right now we're constructing thousands of TileCoordinates per frame because TileCoordinate is const (immutable or not))

Reply:
> I think what @josxha's trying to say is that if you add a non-final attribute to an https://github.com/immutable marked class (for example in another PR), then the linter will immediately get you to reconsider.

Yes, indeed, my intention here was to ensure that if someone wants to add mutable fields in the future in pull requests they will get blocked by a warning. I haven't had polymorphism in mind at all. Now I understand your reason you don't want to add it much better. Sorry for the confusion and thanks for bearing with me.

Closing this this as resolved.